components/proxy/src/main/java/com/hotels/styx/ServerConfigSchema.java

```diff
@@ -30,133 +33,132 @@
-                    optional("originId", object(
-                            field("name", string())
-                    )),
-                    optional("requestId", object(
-                            field("name", string())
-                    )),
-                    atLeastOne("styxInfo", "originId", "requestId")
-            ))
-            .rootSchema(schema(
-                    field("proxy", object(PROXY_CONNECTOR_CONFIG)),
-                    field("admin", object(ADMIN_CONNECTOR_CONFIG)),
-                    field("services", object(
-                            field("factories", map(object(opaque())))
-                    )),
-                    optional("url", object(URL_ENCODING_CONFIG)),
-                    optional("request-logging", object(REQUEST_LOGGING_CONFIG)),
-                    optional("styxHeaders", object(STYX_HEADERS_CONFIG)),
-                    optional("include", string()),
-                    optional("retrypolicy", object(opaque())),
-                    optional("loadBalancing", object(opaque())),
-                    optional("plugins", object(
-                            optional("active", string()),
-                            optional("all", map(object(opaque())))
-                    )),
-                    optional("jvmRouteName", string()),
-                    optional("originRestrictionCookie", string()),
-                    optional("responseInfoHeaderFormat", string()),
-                    optional("httpPipeline", object(opaque())),
-                    optional("logFormat", string()),
-                    optional("userDefined", object(opaque())),
-                    optional("requestTracking", bool()),
-                    optional("routingObjects", object(opaque()))
-            ))
-            .build();
+
+    private static final DocumentFormat.Builder STYX_SERVER_CONFIGURATION_SCHEMA_BUILDER;
+
+    static {
+
+        Schema.FieldValue serverConnectorsSchema = object(
+                optional("http", object(
+                        field("port", integer())
+                )),
+                optional("https", object(
+                        field("port", integer()),
+                        optional("sslProvider", string()),
+                        optional("certificateFile", string()),
+                        optional("certificateKeyFile", string()),
+                        optional("sessionTimeoutMillis", integer()),
+                        optional("sessionCacheSize", integer()),
+                        optional("cipherSuites", list(string())),
+                        optional("protocols", list(string()))
+                )),
+                atLeastOne("http", "https")
+        );
+
+        Schema.FieldValue logFormatSchema = object(
+                optional("enabled", bool()),
+                optional("longFormat", bool()),
+                atLeastOne("enabled", "longFormat")
+        );
+
+        STYX_SERVER_CONFIGURATION_SCHEMA_BUILDER = newDocument()
```

Comment:
We've seen users adding configuration parameters for their own benefit in the styx config; I understand those would break now. If we defined a styx branch/root and ignored the config parameters at the same level than styx in our validation, we would have a quick/dirty fix. Do you think it makes sense to add it?
E.g;
  styxConfig:
      proxy: ...
 customConfig: 
     thiswontfailvalidation:...

Reply:
Well, for this purpose there is:

```
                            optional("userDefined", object(opaque())),
```

Any custom attributes would go to that `userDefined` namespace, which is entirely opaque to the validator.